src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs

```diff
@@ -11,23 +11,27 @@ internal sealed class ProblemDetailsJsonOptionsSetup : IPostConfigureOptions<Jso
 {
     public void PostConfigure(string? name, JsonOptions options)
     {
-        switch (options.SerializerOptions.TypeInfoResolver)
+        var typeInfoResolverChain = options.SerializerOptions.TypeInfoResolverChain;
+        if (typeInfoResolverChain.Count == 0)
```

Comment:
Hmm, I know this isn't new in this PR, but should we instead add ours and the reflection based one in this case?

Reason is because this configure call is specifically to add our ProblemDetailsJsonContext, but now because there aren't any resolvers you also get the reflection based resolver for Problem Details, feels wrong. Especially if the normal use case for getting no resolvers is when trimming is on, which is when you would want our provided context even more.

Reply:
After much off-GitHub discussion, we'd decided on the following approach:

1. `Microsoft.AspNetCore.Http.Json.JsonOptions` sets the `TypeInfoResolver` to `DefaultJsonTypeInfoResolver` at construction-time based on the STJ feature switch allowing Reflection or not. If Reflection isn't allowed, it leaves it null.  (This is how it works today - but based on the current ASP.NET feature switch).
2. `AddProblemDetails()` registers a `Configure<JsonOptions>()` call back, and always adds the `ProblemDetailsJsonContext` to the beginning of the chain at that time

This has the following behaviors:

1. In a "normal" CoreCLR app, you get the chain `{ ProblemDetailsContext, DefaultResolver }`, like you want
2. If someone clears/nulls etc any of this after the fact, the behavior they get is determined by what they changed. At least it is deterministic and understandable.
3. In a trimmed/aot app, we will only add `{ ProblemDetailsContext }`, and it is up to the app to add in their SG JsonContext, which can be done before or after the PD context, as they need
    - When someone doesn't, there will be no JsonTypeInfo for their types, and they will get the same behavior they do today
4. This sets a pattern for other libraries who require the same behavior (adding their own source generated JsonContext to the resolver chain).